semgrep-core/src/engine/Tainting_generic.ml

```diff
@@ -47,61 +46,40 @@ module DataflowY = Dataflow.Make (struct
   let short_string_of_node n = Display_IL.short_string_of_node_kind n.F2.n
 end)
 
-let match_pat_eorig pat =
-  match pat with
-  | [] -> fun _ -> false
-  | xs ->
-      let xs =
-        xs
-        |> List.map (function
-             | AST.E e -> e
-             | AST.S { AST.s = AST.ExprStmt (e, _); _ } ->
-                 (* Some statements in the input language are translated into
-                  * expressions in the Generic AST. This is e.g. the case of
-                  * `echo' in PHP. This small hack allows us to annotate those
-                  * statements as souces/sanitizers/sinks. *)
-                 e
-             | _ ->
-                 failwith "Only Expr patterns are supported in tainting rules")
-      in
-      let pat = Common2.foldl1 (fun x acc -> AST.DisjExpr (x, acc)) xs in
-      fun eorig ->
-        (* the rule is just used by match_e_e for profiling stats *)
-        let rule =
-          {
-            R2.id = "<tainting>";
-            pattern = AST.E pat;
-            pattern_string = "<tainting> pat";
-            message = "";
-            severity = R2.Error;
-            languages = [];
-          }
-        in
-
-        let env =
-          Matching_generic.empty_environment None Config_semgrep.default_config
-        in
-        let matches_with_env = Match_patterns.match_e_e rule pat eorig env in
-        matches_with_env <> []
-
-let match_pat_exp pat exp =
-  let eorig = exp.IL.eorig in
-  match_pat_eorig pat eorig
-
-(*s: function [[Tainting_generic.match_pat_instr]] *)
-let match_pat_instr pat instr =
-  let eorig = instr.IL.iorig in
-  match_pat_eorig pat eorig
-
-(*e: function [[Tainting_generic.match_pat_instr]] *)
+let any_in_ranges any ranges =
+  let tok1, tok2 = Visitor_AST.range_of_any any in
```

Comment:
Maybe add a comment that says "This is potentially slow. We may need to store
range position in the AST at some point".

Reply:
Added!